src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Windows.vb

```diff
@@ -0,0 +1,204 @@
+                        IO.Directory.Delete(FullSourcePath, recursive:=False)
+                    End If
+                End If
+            End If
+
+        End Sub
+
+        '''**************************************************************************
+        ''' ShellDelete
+        ''' <summary>
+        ''' Sets relevant flags on the SHFILEOPSTRUCT and calls into SHFileOperation to delete file / directory.
+        ''' </summary>
+        ''' <param name="FullPath">Full path to the file / directory.</param>
+        ''' <param name="ShowUI">ShowDialogs to display progress and confirmation dialogs. Otherwise HideDialogs.</param>
+        ''' <param name="recycle">SendToRecycleBin to delete to Recycle Bin. Otherwise DeletePermanently.</param>
+        ''' <param name="OnUserCancel">Throw exception or not if the operation was canceled (by user or errors in the system).</param>
+        ''' <remarks>
+        ''' We don't need to consider Recursive flag here since we already verify that in DeleteDirectory.
+        ''' </remarks>
+        Private Shared Sub ShellDelete(ByVal FullPath As String,
+            ByVal ShowUI As UIOptionInternal, ByVal recycle As RecycleOption, ByVal OnUserCancel As UICancelOption, ByVal FileOrDirectory As FileOrDirectory)
+
+            Debug.Assert(FullPath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullPath), "FullPath must be a full path!!!")
+            Debug.Assert(ShowUI <> UIOptionInternal.NoUI, "Why call ShellDelete if ShowUI is NoUI???")
+
+            ' Set fFlags to control the operation details.
+            Dim OperationFlags As ShFileOperationFlags = GetOperationFlags(ShowUI)
+            If (recycle = RecycleOption.SendToRecycleBin) Then
+                OperationFlags = OperationFlags Or ShFileOperationFlags.FOF_ALLOWUNDO
+            End If
+
+            ShellFileOperation(SHFileOperationType.FO_DELETE, OperationFlags, FullPath, Nothing, OnUserCancel, FileOrDirectory)
+        End Sub
+
+        '''**************************************************************************
+        ''' ShellFileOperation
+        ''' <summary>
+        ''' Calls NativeMethods.SHFileOperation with the given SHFILEOPSTRUCT, notifies the shell of change,
+        ''' and throw exceptions if needed.
+        ''' </summary>
+        ''' <param name="OperationType">Value from SHFileOperationType, specifying Copy / Move / Delete</param>
+        ''' <param name="OperationFlags">Value from ShFileOperationFlags, specifying overwrite, recycle bin, etc...</param>
+        ''' <param name="FullSource">The full path to the source.</param>
+        ''' <param name="FullTarget">The full path to the target. Nothing if this is a Delete operation.</param>
+        ''' <param name="OnUserCancel">Value from UICancelOption, specifying to throw or not when user cancels the operation.</param>
+        ''' <remarks>
+        ''' !!!!! SECURITY WARNING !!!!
```

Comment:
Everything to do with System.Security.Permissions can be removed including this comment and any strings constructed or work performed only for that purpose such as CheckPermissionPath below. There may be others, I didn't look.